src/config.rs

```diff
@@ -0,0 +1,399 @@
+            bail!("Cloud region must be either 'US' or 'EU'");
+        }
+
+        let access_token = get_env("INPUT_ACCESS_TOKEN")
+            .ok_or_else(|| anyhow::anyhow!("Access token is required"))?;
+
+        let secrets = get_env("INPUT_SECRETS")
+            .ok_or_else(|| anyhow::anyhow!("Secrets are required"))?
+            .lines()
+            .map(str::trim)
+            .filter(|s| !s.is_empty())
+            .map(String::from)
+            .collect();
+
+        let base_url = get_env("INPUT_BASE_URL");
+        let api_url = get_env("INPUT_API_URL");
+        let identity_url = get_env("INPUT_IDENTITY_URL");
+
+        validate_urls(
+            base_url.as_deref(),
+            api_url.as_deref(),
+            identity_url.as_deref(),
+        )?;
+
+        let set_env = get_env("INPUT_SET_ENV").is_some_and(|val| val != "false");
+
+        Ok(Self {
+            access_token,
+            secrets,
+            cloud_region,
+            base_url,
+            api_url,
+            identity_url,
+            set_env,
+        })
+    }
+}
+
+fn validate_urls(
+    base_url: Option<&str>,
+    api_url: Option<&str>,
+    identity_url: Option<&str>,
+) -> Result<()> {
+    if base_url.is_none() && api_url.is_none() && identity_url.is_none() {
+        return Ok(()); // No URLs provided, nothing to validate
+    }
+
+    if let Some(url) = base_url
+        && !url.starts_with("http://")
+        && !url.starts_with("https://")
+    {
+        bail!("base_url must start with 'https://' or 'http://'");
+    }
+
+    if let Some(url) = api_url
+        && !url.starts_with("http://")
+        && !url.starts_with("https://")
+    {
+        bail!("api_url must start with 'https://' or 'http://'");
+    }
+
+    if let Some(url) = identity_url
+        && !url.starts_with("http://")
+        && !url.starts_with("https://")
+    {
+        bail!("identity_url must start with 'https://' or 'http://'");
+    }
+
+    Ok(())
+}
+
+/// Infers the API and Identity URLs based on the base URL or defaults.
+/// If none of these values are set, it defaults to US or EU URLs,
+/// with US being preferred.
+pub fn infer_urls(config: &Config) -> Result<(String, String)> {
+    let (api_url, identity_url) = match config.cloud_region.to_lowercase().as_str() {
+        // A cloud region was specified; use it
+        "eu" => {
+            debug!("Using EU cloud region URLs");
+            (
+                EU_DEFAULT_API_URL.to_string(),
+                EU_DEFAULT_IDENTITY_URL.to_string(),
+            )
+        }
+        "us" => {
+            debug!("Using US cloud region URLs");
+            (
+                US_DEFAULT_API_URL.to_string(),
+                US_DEFAULT_IDENTITY_URL.to_string(),
+            )
+        }
+
+        // A cloud region was not specified; fall back to inferring the URLs
+        _ => {
+            debug!("No cloud region specified; inferring URLs");
+            let (api_url, identity_url) =
```

Comment:
nit: If you can omit the `let` assignment and therefore the return below that would be nice.